src/listfunc.c

```diff
@@ -1729,6 +1732,64 @@ Obj FuncCOPY_LIST_ENTRIES( Obj self, Obj args )
 
 }
 
+
+Obj FuncLIST_WITH_IDENTICAL_ENTRIES(Obj self, Obj n, Obj obj)
+{
+    if (!IS_NONNEG_INTOBJ(n)) {
+        ErrorQuit("<n> must be a non-negative integer (not a %s)",
+                  (Int)TNAM_OBJ(n), 0L);
+    }
+
+    Obj  list = 0;
+    Int  len = INT_INTOBJ(n);
+    UInt tnum = TNUM_OBJ(obj);
+
+    if (tnum == T_CHAR) {
+        UChar c = *(UChar *)ADDR_OBJ(obj);
```

Comment:
This should be replaced by `UInt c = CHAR_VALUE(obj);` (when I wrote this code, `CHAR_VALUE` did not yet exist).
Of course you can also get rid of the intermediate variable `c` and change the `memset` below to take `CHAR_VALUE(obj)` as argument.